### PR TITLE
Fixed typos and update documentation

### PR DESCRIPTION
**Pull request by karan6181**
## Description
- Fixed typos and update documentation

## Issue
- Fixes: CO-1241

### CODE REVIEW

File: docs/source/getting_started/user_guide.md [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
-- {class}`streaming.TSVWriter`: Writes the dataset into `.tsv` (Tab Separated Values) extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which converts the data from that format to string and vice-versa.
-- {class}`streaming.XSVWriter`: Writes the dataset into `.xsv` (user defined Separated Values) extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which converts the data from that format to string and vice-versa.
+- {class}`streaming.TSVWriter`: Writes the dataset into `.tsv` (Tab Separated Values) extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which convert the data from that format to string and vice-versa.
+- {class}`streaming.XSVWriter`: Writes the dataset into `.xsv` (user defined Separated Values) extension. It supports various encoding/decoding formats(`str`, `int`, and `float`) which convert the data from that format to string and vice-versa.
 
-For more information about writers and its parameters, look at the [API reference doc](../api_reference/).
+For more information about writer and their parameters, look at the [API reference doc](../api_reference/streaming.rst).
```
Comment (knighton) [inline docs/source/getting_started/user_guide.md:12]: writers/their

File: streaming/text/c4.py [hunk 2]
```diff
@@ -22,24 +22,24 @@ class C4(Dataset):
...
+            downloaded shards. If set to None, keep iff remote is local. Defaults to ``None``.
         retry (int): Number of download re-attempts before giving up. Defaults to ``2``.
-        timeout (float): Number of seconds to wait for a shard to download before raising an
-            exception. Defaults to ``60``.
-        hash (str, optional): Optional hash or checksum algorithm to use to validate shards.
+        timeout (float): Number of seconds to wait for a shard to download before raising
+            an exception. Defaults to ``60``.
+        hash (str, optional): Hash or checksum algorithm to use to validate shards. Defaults to
+            ``None``.
+        batch_size (int, optional): Batch size that will be used on each device's DataLoader.
```
Comment (knighton) [inline streaming/text/c4.py:41]: may want to leave in the "hinting" part in, as the field is optional

File: streaming/text/enwiki.py [hunk 3]
```diff
@@ -14,24 +14,24 @@ class EnWiki(Dataset):
...
-        shuffle (bool): Whether to shuffle the samples while iterating. Defaults to ``True``.
-        prefetch (int, optional): Target number of samples remaining to prefetch while iterating.
+        local (str): Local filesystem directory where dataset is cached during operation.
+        remote (str, optional): Remote directory (S3 or local filesystem) where dataset is stored.
             Defaults to ``None``.
-        keep_zip (bool, optional): Whether to keep or delete the compressed file when
-            decompressing downloaded shards. If set to None, keep iff remote is local. Defaults to
+        split (str, optional): The dataset split to use, either 'train' or 'val'. Defaults to
             ``None``.
+        shuffle (bool): Whether to shuffle the samples in this dataset. Defaults to
```
Comment (knighton) [inline streaming/text/enwiki.py:22]: same commentary as above

File: streaming/vision/ade20k.py [hunk 4]
```diff
@@ -16,24 +16,24 @@ class ADE20K(Dataset):
...
-        shuffle (bool): Whether to shuffle the samples while iterating. Defaults to ``True``.
-        prefetch (int, optional): Target number of samples remaining to prefetch while iterating.
+        local (str): Local filesystem directory where dataset is cached during operation.
+        remote (str, optional): Remote directory (S3 or local filesystem) where dataset is stored.
             Defaults to ``None``.
-        keep_zip (bool, optional): Whether to keep or delete the compressed file when
-            decompressing downloaded shards. If set to None, keep iff remote is local. Defaults to
+        split (str, optional): The dataset split to use, either 'train' or 'val'. Defaults to
             ``None``.
+        shuffle (bool): Whether to shuffle the samples in this dataset. Defaults to
```
Comment (knighton) [inline streaming/vision/ade20k.py:24]: same commentary as above

File: streaming/text/c4.py [hunk 5]
```diff
@@ -22,24 +22,24 @@ class C4(Dataset):
...
-            ``None``.
-        split (str, optional): Which dataset split to use, if any. Defaults to ``None``.
-        shuffle (bool): Whether to shuffle the samples while iterating. Defaults to ``True``.
-        prefetch (int, optional): Target number of samples remaining to prefetch while iterating.
+        local (str): Local filesystem directory where dataset is cached during operation.
+        remote (str, optional): Remote directory (S3 or local filesystem) where dataset is stored.
             Defaults to ``None``.
-        keep_zip (bool, optional): Whether to keep or delete the compressed file when
-            decompressing downloaded shards. If set to None, keep iff remote is local. Defaults to
+        split (str, optional): The dataset split to use, either 'train' or 'val'. Defaults to
```
[the remark this answers is not in the document]
Comment (knighton) [reply streaming/text/c4.py:28]: oh nevermind you are correct, i am still waking up this morning

File: streaming/vision/coco.py [hunk 6]
```diff
@@ -21,14 +21,14 @@ class COCO(Dataset):
             Defaults to ``None``.
         split (str, optional): The dataset split to use, either 'train' or 'val'. Defaults to
             ``None``.
-        shuffle (bool, optional): Whether to shuffle the train samples in this dataset. Defaults to
+        shuffle (bool, optional):Whether to shuffle the samples in this dataset. Defaults to
```
Comment (knighton) [inline streaming/vision/coco.py:24]: missing space

**Review by knighton — APPROVED** [review]
Some pedantry

File: streaming/text/c4.py [hunk 7]
```diff
@@ -22,24 +22,24 @@ class C4(Dataset):
...
             ``None``.
+        shuffle (bool): Whether to shuffle the samples in this dataset. Defaults to
```
[the remark this answers is not in the document]
Comment (knighton) [reply streaming/text/c4.py:30]: "Whether to iterate over the samples in randomized order"?

File: docs/source/getting_started/user_guide.md (hunk 1 above)
Comment (karan6181) [reply docs/source/getting_started/user_guide.md]: Fixed in https://github.com/mosaicml/streaming/pull/42/commits/c023c30b7f55b89dd9f6526480c0c5a272940bba.

File: streaming/text/c4.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (karan6181) [reply streaming/text/c4.py]: Fixed in https://github.com/mosaicml/streaming/pull/42/commits/c023c30b7f55b89dd9f6526480c0c5a272940bba.

File: streaming/text/c4.py (hunk 2 above)
Comment (karan6181) [reply streaming/text/c4.py]: Fixed in https://github.com/mosaicml/streaming/pull/42/commits/c023c30b7f55b89dd9f6526480c0c5a272940bba.

File: streaming/text/enwiki.py (hunk 3 above)
Comment (karan6181) [reply streaming/text/enwiki.py]: Fixed in https://github.com/mosaicml/streaming/pull/42/commits/c023c30b7f55b89dd9f6526480c0c5a272940bba.

File: streaming/vision/ade20k.py (hunk 4 above)
Comment (karan6181) [reply streaming/vision/ade20k.py]: Fixed in https://github.com/mosaicml/streaming/pull/42/commits/c023c30b7f55b89dd9f6526480c0c5a272940bba.

File: streaming/vision/coco.py (hunk 6 above)
Comment (karan6181) [reply streaming/vision/coco.py]: Fixed in https://github.com/mosaicml/streaming/pull/42/commits/c023c30b7f55b89dd9f6526480c0c5a272940bba.

**Pull request merged**